File: src/code-samples/langchain/return-a-string.ts
// :snippet-start: tool-return-values
import { tool } from "langchain";
import * as z from "zod";

const getWeather = tool(({ city }) => `It is currently sunny in ${city}.`, {
  name: "get_weather",
  description: "Get weather for a city.",
  schema: z.object({ city: z.string() }),
});
// :snippet-end:

// :remove-start:
async function main() {
  const result = await getWeather.invoke({ city: "San Francisco" });
  if (result !== "It is currently sunny in San Francisco.") {
    throw new Error(
      `Expected "It is currently sunny in San Francisco.", got "${result}"`,
    );
  }
  console.log("✓ Tool works as expected");
}
main();
// :remove-end:

File: src/code-samples/langchain/return-a-string.py
# :snippet-start: tool-return-values-py
from langchain.tools import tool


@tool
def get_weather(city: str) -> str:
    """Get weather for a city."""
    return f"It is currently sunny in {city}."


# :snippet-end:

# :remove-start:
if __name__ == "__main__":
    result = get_weather.invoke({"city": "San Francisco"})
    assert result == "It is currently sunny in San Francisco."
    print("✓ Tool works as expected")
# :remove-end:
